packages/react/src/components/Toolbar/ToolbarCustomItem.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+
+import {
+  ChildrenComponentProps,
+  ContentComponentProps,
+  createShorthandFactory,
+  UIComponentProps,
+  UIComponent,
+  childrenExist,
+  commonPropTypes,
+  isFromKeyboard,
+} from '../../lib'
+
+import { ComponentEventHandler, WithAsProp, withSafeTypeForAs } from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { defaultBehavior } from '../../lib/accessibility'
+import { IS_FOCUSABLE_ATTRIBUTE } from '../../lib/accessibility/FocusZone'
+import * as _ from 'lodash'
+
+export interface ToolbarCustomItemProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   */
+  accessibility?: Accessibility
+
+  /** A custom item can remove element padding, vertically or horizontally. */
+  fitted?: boolean | 'horizontally' | 'vertically'
+
+  /** A custom item can be focused. */
+  focusable?: boolean
+
+  /** A custom item can't be actionable. */
+  onClick: never
+
+  /**
+   * Called after user's focus. Will be called only if the item is focusable.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onFocus?: ComponentEventHandler<ToolbarCustomItemProps>
+
+  /**
+   * Called after item blur. Will be called only if the item is focusable.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onBlur?: ComponentEventHandler<ToolbarCustomItemProps>
+}
+
+interface ToolbarCustomItemState {
+  isFromKeyboard: boolean
+}
+
+class ToolbarCustomItem extends UIComponent<
+  WithAsProp<ToolbarCustomItemProps>,
+  ToolbarCustomItemState
+> {
+  static displayName = 'ToolbarCustomItem'
+
+  static className = 'ui-toolbar__customitem'
```

Comment:
`ui-toolbar__custom-item`?

Reply:
We have a UT for this one:

```tsx
test(`is a static equal to "${info.componentClassName}"`, () => {
```

It expects to have: `ui-toolbar__customitem`